### DISCUSSION TITLE
Flux.skip and Flux.take

### DISCUSSION THREAD

**Discussion by KarlHakanNordgren** [micronaut-projects/micronaut-data#1626] [Q&A]
Hi! Do the above two methods push the work down to the repository? (Or do the items get pulled from the repository and then get skipped in the caller?).

Thanks!

**Comment by KarlHakanNordgren** [comment]
@dstepanov : Sorry to pester you always. Do you know the answer to this one?

**Reply by dstepanov** [reply]
They will get skipped.

**Reply by KarlHakanNordgren** [reply]
Thanks for the response! Where will they get skipped -- will the skipping be pushed back to MongoDB, or will the elements be taken out of MongoDB and skipped in the service which made the query?

**Reply by dstepanov** [reply]
They will be fetched and ignored.

**Reply by KarlHakanNordgren** [reply]
Gotcha. Thanks!

**Reply by KarlHakanNordgren** [reply]
Sorry one more (potentially a bit silly) question: If I do `Flux.take(4)`, it does not fetch records 5, 6, 7, etc. correct? (Is there documentation for this or can you point me to the place in the source so I can have a look?)

**Reply by dstepanov** [reply]
I think it will fetch everything or at least some internal batch of records. Are we talking about R2DBC backend, right? 
Technically, Reactor might ask just for 4 records and then close the subscriber, then its up to how upstream is implemented.

**Reply by KarlHakanNordgren** [reply]
The backend is MongoDB. Does the above hold true for MongoDB?

**Reply by dstepanov** [reply]
It depends on how the driver is implemented, but I would assume it will be the same. You should use `skip` and `limit` when fetching results instead of reactive skip operators.
